OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -425,12 +446,18 @@ void AddAsset(ScrollPanelWidget list, string filepath, IReadOnlyPackage package,
 
 			item.IsVisible = () =>
 			{
+				var allowed = (assetTypesToDisplay.HasFlag(AssetType.Sprite) && allowedSpriteExtensions.Any(ext => filepath.EndsWith(ext, true, CultureInfo.InvariantCulture)))
```

Comment:
If I'm reading this correctly, this only changes when `assetTypesToDisplay` changes, right? Then we might want to set the boolean when that changes instead of running this every render(?) when determining visibility.

Reply:
This is per asset (it's in `AddAsset`) and it does so by depending on `filepath` further into the line.